### PR TITLE
RDM-6427 Functional tests for Retrieve search input details for dynamic display

### PR DESCRIPTION
**Pull request by bmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-6427

### Change description ###

Added functional tests for the Retrieve search input details endpoint

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/aat/resources/features/F-063/F-063.feature [hunk 1]
```diff
@@ -0,0 +1,45 @@
+@F-063
+Feature: F-063: Retrieve search input details for dynamic display
+
+  Background:
+    Given an appropriate test context as detailed in the test data source
+
+  @S-215
+  Scenario: should retrieve search inputs
```
Comment (nitishsingla1234) [inline src/aat/resources/features/F-063/F-063.feature]: "should retrieve search inputs for Dynamic display"

File: src/aat/resources/features/F-063/F-063.feature [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    Given a user with [an active profile in CCD]
+    When a request is prepared with appropriate values
+    And the request [contains a valid case type]
+    And it is submitted to call the [Retrieve search input details for dynamic display] operation of [CCD Data Store]
+    Then a positive response is received
+    And the response [contains the correct search inputs for the given case type, along with an HTTP 200 OK]
```
Comment (nitishsingla1234) [inline src/aat/resources/features/F-063/F-063.feature]: [contains the correct search **results** for the given case type, along with an HTTP 200 OK]

File: src/aat/resources/features/F-063/F-063_Base_Test_Data.td.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	"users": {
+    "invokingUser": {
+      "username": "[[$CCD_CASEWORKER_AUTOTEST_EMAIL]]",
+      "password": "[[$CCD_CASEWORKER_AUTOTEST_PASSWORD]]"
+    }
+	},
+
+	"request": {
+		"headers": {
+			"_extends_": "Common_Request_Headers",
```
Comment (nitishsingla1234) [inline src/aat/resources/features/F-063/F-063_Base_Test_Data.td.json]: Do we need the following header with the endpoint?
public static final String UI_SEARCH_INPUT_DETAILS =
            "application/vnd.uk.gov.hmcts.ccd-data-store-api.ui-search-input-details.v2+json;charset=UTF-8";

Comment (nitishsingla1234) [reply src/aat/resources/features/F-063/F-063_Base_Test_Data.td.json]: I believe we are checking the following endpoint.

		path = "/case-types/{caseTypeId}/search-inputs",

Comment (bmcts) [reply src/aat/resources/features/F-063/F-063_Base_Test_Data.td.json]: it is included as a response header in S-215.td.json as it is only relevant for the happy path scenario

File: src/aat/resources/features/F-063/F-063.feature (hunk 2 above)
Comment (bmcts) [reply src/aat/resources/features/F-063/F-063.feature]: it isn't the search results that are returned, it is a list of search inputs that are allowed for that case type... in other words, the different aspects of a case type that can be searched for (eg. defendant name, solicitor name, etc.)

Comment (nitishsingla1234) [reply src/aat/resources/features/F-063/F-063.feature]: Makes sense.

File: src/aat/resources/features/F-063/F-063_Base_Test_Data.td.json (hunk 3 above)
Comment (nitishsingla1234) [reply src/aat/resources/features/F-063/F-063_Base_Test_Data.td.json]: okay....

File: src/aat/resources/features/F-063/F-063.feature (hunk 1 above)
Comment (bmcts) [reply src/aat/resources/features/F-063/F-063.feature]: updated here and in Jira ticket 👍

File: src/aat/resources/features/F-063/S-561.td.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  "specs": [
+    "an active profile in CCD",
+    "contains a case type that does not exist",
+    "contains a HTTP 404 Not Found"
+  ],
+
+  "request": {
+    "pathVariables": {
+      "ctid": "abc"
```
Comment (shrivsksavit) [inline src/aat/resources/features/F-063/S-561.td.json]: as 'abc' might be valid can we put 'non-existing case type ID' as suggested in the past to me as well

File: src/aat/resources/features/F-063/S-214.td.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    }
+  },
+
+  "expectedResponse": {
+    "_extends_": "Common_404_Response",
+    "headers": {
+      "Content-Encoding": "gzip",
+      "Content-Length": "[[ANY]]",
+      "Content-Type": "application/json;charset=UTF-8"
+    },
```
Comment (shrivsksavit) [inline src/aat/resources/features/F-063/S-214.td.json]: can we move "Content-Length": "[[ANY]]",
      "Content-Type": "application/json;charset=UTF-8" to base file

File: src/aat/resources/features/F-063/S-561.td.json (hunk 4 above)
Comment (bmcts) [reply src/aat/resources/features/F-063/S-561.td.json]: updated to "non-existing case type" 👍

File: src/aat/resources/features/F-063/S-214.td.json (hunk 5 above)
Comment (bmcts) [reply src/aat/resources/features/F-063/S-214.td.json]: moved to base test data file 👍

**Pull request merged**